cpp/src/lists/segmented_sort.cu

```diff
@@ -152,7 +152,7 @@ struct SegmentedSortColumn {
   }
 
   template <typename T>
-  std::enable_if_t<is_numeric<T>(), std::unique_ptr<column>> operator()(
+  std::enable_if_t<std::is_integral_v<T>, std::unique_ptr<column>> operator()(
```

Comment:
Should decimal types need to be sorted with radix sort? Does this need to be dispatched based on the "representation" type?

Reply:
I think you are right. Fixed-point should work with radix sort too. 
Do you think that should be added in this PR? 
I think most of the work would be adding gtests and perhaps a benchmark.

Reply:
I think the scope of this PR could be interpreted broadly as "using radix sort when applicable." I would be happiest with adopting the exact check used by the other radix sort condition: `cudf::is_fixed_width<T>() && !cudf::is_floating_point<T>();` and perhaps even naming that with a `constexpr` function as is done in `sort_column.cu`.

https://github.com/rapidsai/cudf/blob/972708afcfaca006a8483a133dfacd540f232ef1/cpp/src/sort/sort_column.cu#L34

I hope that we have some kind of typed test coverage that would cover this case already, but if not then adding a test would be ideal. I don't want to make this overly cumbersome or expand the scope too greatly, but it seems like expanding the scope to "using radix sort when applicable" would improve consistency with `sort_column.cu` as well as improve correctness for the identified bug.

Reply:
Created issue https://github.com/rapidsai/cudf/issues/11729 to align the two implementations.